enterprise/models.py

```diff
@@ -541,3 +541,81 @@ def __repr__(self):
         Return uniquely identifying string representation.
         """
         return self.__str__()
+
+
+@python_2_unicode_compatible
+class EnterpriseCourseEnrollment(TimeStampedModel):
+    """
+    Store information about the enrollment of a user in a course.
+
+    This model is the central source of truth for information about
+    whether a particular user, linked to a particular EnterpriseCustomer,
+    has been enrolled in a course, and is the repository for any other
+    relevant metadata about such an enrollment.
+    """
+
+    class Meta(object):
+        unique_together = (('user', 'course_id',),)
+        app_label = 'enterprise'
+
+    user = models.ForeignKey(
+        EnterpriseCustomerUser,
+        blank=False,
+        null=False,
+        related_name='enterprise_enrollments',
+        help_text=_(
+            "The enterprise learner to which this enrollment is attached."
+        )
+    )
+    consent_granted = models.NullBooleanField(
+        help_text=_(
+            "Whether the learner has granted consent for this particular course."
+        )
+    )
+    course_id = models.CharField(
```

Comment:
Given that the course_id is still unique, and that the `EnterpriseCustomerUser` already contains a link to a particular EdX user, I felt it was better to use the more descriptive identifier. This is especially the case because if we went with the `pk` of the `CourseEnrollment`, we'd be passing back and forth data without real semantic meaning, making interpretation of the database more difficult. In other words, to get meaningful data about `EnterpriseCourseEnrollment`s from the Enterprise app, we'd have to make a large number of requests to the LMS database to determine the course ID each CourseEnrollment ID matched up with.